gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -136,6 +137,13 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
             kspExtension.apOptions.forEach {
                 options += SubpluginOption("apoption", "${it.key}=${it.value}")
             }
+            kspExtension.commandLineArgumentProviders.forEach {
+                val argument = it.asArguments().joinToString("")
+                if (!argument.matches(Regex("/S=/S"))) {
```

Comment:
Pardon me if this is a stupid question, but I cannot find what `/S` mean. Kotlin's [Regex doc](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/-regex/) says that it is [java.util.regex.Pattern](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html) but I can't find it in the doc.

Can we replace the check with `if ('=' !in argument) throw ...`?

Reply:
I think the intention here was to match anything except whitespace (`\S`), but the regex isn't quite right - it should be something more like `"\\S+=\\S+"`.

https://pl.kotl.in/pItEH8Yd8